lineapy/plugins/argo_pipeline_writer.py

```diff
@@ -0,0 +1,244 @@
+
+from kubernetes import client, config
+from typing_extensions import TypedDict
+
+from lineapy.plugins.base_pipeline_writer import BasePipelineWriter
+from lineapy.plugins.task import (
+    DagTaskBreakdown,
+    TaskDefinition,
+    TaskGraph,
+    TaskSerializer,
+    render_task_io_serialize_blocks,
+)
+from lineapy.plugins.taskgen import get_task_definitions
+from lineapy.plugins.utils import load_plugin_template
+from lineapy.utils.logging_config import configure_logging
+from lineapy.utils.utils import prettify
+
+logger = logging.getLogger(__name__)
+configure_logging()
+
+
+class ARGODagFlavor(Enum):
+    PythonOperatorPerSession = 1
+    PythonOperatorPerArtifact = 2
+    # KubernetesPodOperatorPerSession = 3
+    # KubernetesPodOperatorPerArtifact = 4
+
+
+ARGODAGConfig = TypedDict(
+    "ARGODAGConfig",
+    {
+        "namespace": str,
+        "host": str,
+        "verify_ssl": str,
+        "workflow_name": str,
+        "service_account": int,
+        "kube_config": str,
+        "task_serialization": str,
+        "dag_flavor": str,
+    },
+    total=False,
+)
+
+
+class ARGOPipelineWriter(BasePipelineWriter):
+    """
+    Class for pipeline file writer. Corresponds to "ARGO" framework.
+    """
+
+    @property
+    def docker_template_name(self) -> str:
+        return "argo_dockerfile.jinja"
+
+    def _write_dag(self) -> None:
+
+        # Check if the given DAG flavor is a supported/valid one
+        try:
+            dag_flavor = ARGODagFlavor[
+                self.dag_config.get("dag_flavor", "PythonOperatorPerSession")
+            ]
+        except KeyError:
+            raise ValueError(f'"{dag_flavor}" is an invalid ARGO dag flavor.')
+
+        try:
+            task_serialization = TaskSerializer[
+                self.dag_config.get("task_serialization", "LocalPickleArgo")
+            ]
+        except KeyError:
+            raise ValueError(
+                f'"{task_serialization}" is an invalid type of task serialization scheme.'
+            )
+
+        # Construct DAG text for the given flavor
+        full_code = self._write_operators(dag_flavor, task_serialization)
+
+        print(full_code)
+
+        # Write out file
+        file = self.output_dir / f"{self.pipeline_name}_dag.py"
+        file.write_text(prettify(full_code))
+        logger.info(f"Generated DAG file: {file}")
+
+    def _write_operators(
+        self,
+        dag_flavor: ARGODagFlavor,
+        task_serialization: TaskSerializer,
+    ) -> str:
+
+        DAG_TEMPLATE = load_plugin_template("argo_dag.jinja")
+
+        if dag_flavor == ARGODagFlavor.PythonOperatorPerSession:
+            task_breakdown = DagTaskBreakdown.TaskPerSession
+        elif dag_flavor == ARGODagFlavor.PythonOperatorPerArtifact:
+            task_breakdown = DagTaskBreakdown.TaskPerArtifact
+
+        # Get task definitions based on dag_flavor
+        task_defs: Dict[str, TaskDefinition] = get_task_definitions(
```

Comment:
Few things here.

Before we merge this PR, make sure to pull from main. Then a few asks for you in the following sections. 

1. get_task_definitions will change to [get_task_graph](https://github.com/LineaLabs/lineapy/blob/6aee596d7b0a634adb77a3973307026b110a012d/lineapy/plugins/kubeflow_pipeline_writer.py#L78) and return the task_graph so you don't need to write it. 
2. Can you add a no-op setup and teardown [like so](https://github.com/LineaLabs/lineapy/blob/6aee596d7b0a634adb77a3973307026b110a012d/lineapy/plugins/kubeflow_pipeline_writer.py#L84)

Reply:
Got it! Added all!